### PR TITLE
Added option to randomize INITIAL_DAY

### PR DESCRIPTION
**Pull request by ashGlaw**
#### Summary
```SUMMARY: Features "Added option to randomize INITIAL_DAY"```

#### Purpose of change
I've been told by a number of people that when they create their worlds they find a random number generator online and pick a random start date. It wouldn't be hard to add that to the options and save them some time.

#### Describe the solution
Setting the INITIAL_DAY option to -1 has been assigned to choose a random start date. The range of options is equal to 4x the SEASON_LENGTH, minus 1.

#### Describe alternatives you've considered
Instead of setting the random value to -1 a new boolean option could be created, and when set to true would grey out the INITIAL_DAY options.

Could pull the season length earlier to avoid having to retrieve it twice

#### Testing
Compiled and ran the game with a new world. Set the INITIAL_DAY to -1. Started on day 69 of summer. Tried again and got Day 38 in winter. No errors reported.

#### Additional context
I'm open to changing how this is done if it would help

### CODE REVIEW

**Comment by Funguss** [comment]
YESSSSS! I was just thinking the other day how in so many games with an element of farming or some kind of survival aspect you always start at the beginning of spring. This is going to make it feel a lot more immersive, thank you!

**Comment by ashGlaw** [comment]
Looks like the Travis build has an issue, but I don't know how to parse what it's trying to tell me. I'd love to fix it, though

File: src/game.cpp [hunk 1]
```diff
@@ -11456,7 +11456,11 @@ void game::start_calendar()
         }
     } else {
         // No scenario, so use the starting date+time configured in world options
-        const int initial_days = get_option<int>( "INITIAL_DAY" );
+        int initial_days = get_option<int>( "INITIAL_DAY" );
+        if( initial_days == -1 ) {
+            // 0 - 363 for a 91 day season
+            initial_days = rng( 0, ( get_option<int>( "SEASON_LENGTH" ) * 4 ) - 1 );
```
Comment (AMurkin) [inline src/game.cpp]:
```suggestion
            initial_days = rng( 0, get_option<int>( "SEASON_LENGTH" ) * 4 - 1 );
```

**Comment by jkraybill** [comment]
As a full-random player, thanks!! Will add more variability to starting scenarios. Being able to set range of the randomizer, and/or allowing scenarios to do the same, would be a great future addition. Having current scenario starts non-dynamic is an obstacle to json-based scenario expansions.

File: src/options.cpp [hunk 2]
```diff
@@ -2057,8 +2057,8 @@ void options_manager::add_options_world_default()
        );
 
     add( "INITIAL_DAY", "world_default", translate_marker( "Initial day" ),
-         translate_marker( "How many days into the year the cataclysm occurred.  Day 0 is Spring 1.  Can be overridden by scenarios.  This does not advance food rot or monster evolution." ),
-         0, 999, 60
+         translate_marker( "How many days into the year the cataclysm occurred.  Day 0 is Spring 1. Day -1 randomizes the start date. Can be overridden by scenarios.  This does not advance food rot or monster evolution." ),
```
Comment (BevapDin) [inline src/options.cpp]:
```suggestion
         translate_marker( "How many days into the year the cataclysm occurred.  Day 0 is Spring 1.  Day -1 randomizes the start date. Can be overridden by scenarios.  This does not advance food rot or monster evolution." ),
```
We require two spaces between sentences.

**Comment by ZhilkinSerg** [comment]
I would prefer a new Boolean option instead, but it is up to @kevingranade to decide for new options.

**Comment by ashGlaw** [comment]
Thanks for the tags. Is that something you need special privileges to do? I haven't run across the option to tag my PRs

**Comment by ashGlaw** [comment]
> 
> 
> I would prefer a new Boolean option instead, but it is up to @kevingranade to decide for new options.

Thanks for the at.
I could probably do a Boolean option, if that's how it's preferred, but this seemed simple and elegant and required fewer changes.

**Pull request merged**